controller/src/main/java/org/jboss/as/controller/extension/SubsystemInformation.java

```diff
@@ -56,4 +58,25 @@
      * @return the micro interface version, or {@code null} if the subsystem does not have a versioned interface
      */
     Integer getManagementInterfaceMicroVersion();
+
+
+    /**
+     * Gets full version of the subsystem
+     * @return full {@link ModelVersion}
+     */
+    default ModelVersion getManagementInterfaceVersion(){
+        return ModelVersion.create(getManagementInterfaceMajorVersion() != null ? getManagementInterfaceMajorVersion() : 0,
+                getManagementInterfaceMinorVersion() != null ? getManagementInterfaceMinorVersion() : 0,
+                getManagementInterfaceMicroVersion() != null ? getManagementInterfaceMicroVersion() : 0
+        );
+    }
+
+
+    default boolean isDeprecated(){
+        return false;
+    }
+
+    default boolean isHostCapable() {
+                return false;
+            }
```

Comment:
There was a reason I added that, but I cannot remember what it was. 
If and when I remember again I can send separate PR to do that.